apps/arweave/src/ar_chunk_storage.erl

```diff
@@ -871,6 +873,29 @@ is_offset_valid(Byte, BucketStart, ChunkOffset) ->
 	Delta = Byte - (BucketStart + ChunkOffset rem ?DATA_CHUNK_SIZE),
 	Delta >= 0 andalso Delta < ?DATA_CHUNK_SIZE.
 
+filter_by_sync_record(Chunks, _Byte, _Start, _ChunkFileStart, _StoreID, 1) ->
+	Chunks;
+filter_by_sync_record([], _Byte, _Start, _ChunkFileStart, _StoreID, _ChunkCount) ->
+	[];
+filter_by_sync_record([{PaddedEndOffset, Chunk} | Rest], Byte, Start, ChunkFileStart, StoreID, ChunkCount) ->
+	case ar_sync_record:is_recorded(PaddedEndOffset, ar_chunk_storage, StoreID) of
+		false ->
+			%% Filter out repacking artifacts when reading a range of chunks.
+			%% This warning may also occur due to unlucky timing when the chunk is
+			%% being removed.
+			?LOG_WARNING([{event, found_chunk_not_in_sync_record},
```

Comment:
what do you think about invalidating the chunk when this occurs? so it can be resynced?

Reply:
I do not think resyncing is blocked in this case - we are not looking into the ar_data_sync sync_record here. Another possibility may be the chunk is synced and stored outside of ar_chunk_storage.